test/hotspot/jtreg/runtime/cds/appcds/methodHandles/JDKMethodHandlesTestRunner.java

```diff
@@ -25,19 +25,91 @@
+    private static final String ps = System.getProperty("path.separator");
+    private static final String testPackageName = "test.java.lang.invoke";
+
+    public static void test(String testClassName) throws Exception {
+        String appJar = JarBuilder.build("MH", new File(classDir), null);
+        String classList = testClassName + ".list";
+        String archiveName = testClassName + ".jsa";
+        // Disable VerifyDpendencies when running with debug build because
+        // the test requires a lot more time to execute with the option enabled.
+        String verifyOpt =
+            Platform.isDebugBuild() ? "-XX:-VerifyDependencies" : "-showversion";
+
+        String junitJar = Path.of(Test.class.getProtectionDomain().getCodeSource().getLocation().toURI()).toString();
+
+        String jars = appJar + ps + junitJar;
+
+        CDSAppTester tester = new CDSAppTester(testClassName) {
+                @Override
+                public String classpath(RunMode runMode) {
+                    return jars;
+                }
+
+                @Override
+                public String[] vmArgs(RunMode runMode) {
+                    if (runMode.isProductionRun()) {
+                        return new String[] {
+                            "-Xlog:class+load,cds=debug",
+                            verifyOpt,
+                        };
+                    } else {
+                        return new String[] {
+                            verifyOpt,
+                        };
+                    }
+                }
+
+                @Override
+                public String[] appCommandLine(RunMode runMode) {
+                    return new String[] {
+                        mainClass,
+                        testPackageName + "." + testClassName,
+                    };
+                }
+
+                @Override
+                public void checkExecution(OutputAnalyzer out, RunMode runMode) throws Exception {
+                    out.shouldHaveExitValue(0);
+                    if (runMode.isProductionRun()) {
+                        out.shouldMatch(".class.load. test.java.lang.invoke." + testClassName +
+                                        "[$][$]Lambda.*/0x.*source:.*shared.*objects.*file");
+                    }
+                }
+            };
+
+        String workflow = System.getProperty("cds.app.tester.workflow");
+        tester.run(workflow);
+    }
+}
+
+class TestMHApp {
     public static void main(String args[]) throws Exception {
         try {
             Class<?> testClass = Class.forName(args[0]);
             System.out.println(testClass);
             Object obj = testClass.newInstance();
             final List<Method> allMethods = new ArrayList<Method>(Arrays.asList(testClass.getDeclaredMethods()));
             for (final Method method : allMethods) {
-                //System.out.println(method.toString());
```

Comment:
Pre-existing:
Can you also remove the comment `System.out.println` at line 138?

Reply:
Fixed.